File: app/src/guide-table/api/code/php/example2.php
/* 3. Replacing a text and saving it in another variable */

$regex  = '/\d([a-z])/';
$replacement = '@';
$string = '1a2b3c';

$newString = preg_replace($regex, $replacement, $string);
echo $newString; // @@@

/* 4. The same but this time adding a limit */

$limit = 1;
$newString = preg_replace($regex, $replacement, $string, $limit);
echo $newString; // @2b3c

/* 5. Splitting a string into an array */

$regex = '/\d/';
$array = preg_split($regex, $string);
print_r($array);

/*
    Array (
        [0] => 
        [1] => a
        [2] => b
        [3] => c
    )
*/

File: app/src/guide-table/api/code/java/example2.java
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class Main {
  public static void main(String[] args) {
    
    /* 4. Replacing only the first occurrence */
        
    String regex  = "\\d([a-z])";
    String replacement = "@";
    String string = "1a2b3c";

    Pattern p = Pattern.compile(regex);
    Matcher m = p.matcher(string);
    
    String newString = m.replaceFirst(replacement);
    System.out.println(newString); // @2b3c

    /* 5. Replacing all occurrences */
        
    String regex  = "\\d([a-z])";
    String replacement = "@";
    String string = "1a2b3c";

    Pattern p = Pattern.compile(regex);
    Matcher m = p.matcher(string);
    
    String newString = m.replaceAll(replacement);
    System.out.println(newString); // @@@

    /* 6. Splitting a string into an array */

    String regex  = "\\d";
    String string = "1a2b3c";

    Pattern p = Pattern.compile(regex);
    
    String[] arrayString = p.split(string);
    
    for (String s : arrayString) {
        System.out.print(s + " "); // a b c
    }
  }
}
